darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterStrategyFactoryImpl.java

```diff
@@ -109,18 +109,35 @@ public DarkClusterStrategy getOrCreate(@Nonnull String darkClusterName, @Nonnull
    */
   private DarkClusterStrategy createStrategy(String darkClusterName, DarkClusterConfig darkClusterConfig)
   {
-    if (darkClusterConfig.hasMultiplier() && darkClusterConfig.getMultiplier() > 0)
+    if (darkClusterConfig.hasMultiplierStrategyList())
     {
-      BaseDarkClusterDispatcher baseDarkClusterDispatcher = new BaseDarkClusterDispatcherImpl(darkClusterName, _darkClusterDispatcher,
-                                                                                              _notifier, _verifierManager);
-      return new RelativeTrafficMultiplierDarkClusterStrategy(_sourceClusterName, darkClusterName, darkClusterConfig.getMultiplier(),
-                                                              baseDarkClusterDispatcher, _notifier, _facilities.getClusterInfoProvider(), _random);
-    }
-    else
-    {
-      // Falling into this clause means that the user does not want traffic to be sent to the dark cluster, temporary or otherwise.
-      return new NoOpDarkClusterStrategy();
+      DarkClusterStrategyNameArray strategyList = darkClusterConfig.getMultiplierStrategyList();
+      for (com.linkedin.d2.DarkClusterStrategyName multiplierStrategyType : strategyList)
+      {
+        switch(multiplierStrategyType)
+        {
+          case RELATIVE_TRAFFIC:
+            if (darkClusterConfig.getMultiplier() <= 0)
+            {
+              // it is a valid case to set RELATIVE_TRAFFIC but set the multiplier to zero to
+              // temporarily set the traffic to zero; don't log anything.
+              return new NoOpDarkClusterStrategy();
+            }
+            BaseDarkClusterDispatcher baseDarkClusterDispatcher =
+              new BaseDarkClusterDispatcherImpl(darkClusterName, _darkClusterDispatcher, _notifier, _verifierManager);
+            return new RelativeTrafficMultiplierDarkClusterStrategy(_sourceClusterName, darkClusterName, darkClusterConfig.getMultiplier(),
+                                                                    baseDarkClusterDispatcher, _notifier, _facilities.getClusterInfoProvider(),
+                                                                    _random);
+          case CONSTANT_QPS:
+            // the constant qps strategy is not yet implemented, continue to the next strategy if it exists
+            break;
+          default:
+            return new NoOpDarkClusterStrategy();
```

Comment:
not necessary since we return no op strategy at end anyway.

Reply:
ok.